app/lib/actions/pulp/repository/presenters/puppet_presenter.rb

```diff
@@ -25,11 +25,14 @@ def humanized_details
           end
 
           def finished_count
-            task_result['finished_count']
+            return 0 unless task_progress_details.key?('modules')
+            task_progress_details['modules']['finished_count']
           end
 
           def total_count
-            task_progress_details['modules']['total_count'] || 0
+            return 0 unless task_progress_details.key?('modules')
+            return 0 if task_progress_details['modules']['total_count'].nil?
```

Comment:
I feel like I lose expressing the fact that due to whatever calls this, sometimes it provides a hash without the modules section, sometimes it provides a hash with the modules section but not total_count key and sometimes it provides a hash with the modules section and the total_count key. Could be I read it that way because I wrote it though as well :)